File: clients/python/examples/find.py
from __future__ import annotations

from sikuligo_client.client import SikuliGrpcClient


def main() -> int:
    client = SikuliGrpcClient()
    try:
        res = client.find_on_screen("assets/pattern.png", exact=True, timeout_millis=3000)
        match = res.match
        print(
            f"match rect=({match.rect.x},{match.rect.y},{match.rect.w},{match.rect.h}) "
            f"score={match.score:.3f} target=({match.target.x},{match.target.y})"
        )
    finally:
        client.close()
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: clients/node/examples/find.ts
import { Screen, Pattern } from "../src";

async function main(): Promise<void> {
  const screen = await Screen.start();
  try {
    const pattern = new Pattern("assets/pattern.png").exact();
    const match = await screen.find(pattern);
    console.log(`found rect=(${match.x},${match.y},${match.w},${match.h}) target=(${match.targetX},${match.targetY})`);
  } finally {
    await screen.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
